packages/api/e2e/amino.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { StdFee } from '@cosmjs/amino/build/signdoc';
+import { Secp256k1HdWallet } from '@cosmjs/amino/build/secp256k1hdwallet';
+import { MsgCreate } from '../src/generated/regen/ecocredit/basket/v1/tx';
+
+const TEST_ADDRESS = 'regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp';
+const REDWOOD_NODE_TM_URL = 'http://redwood.regen.network:26657/';
+const TEST_FEE: StdFee = {
+  amount: [
+    {
+      denom: 'uregen',
+      amount: '5000',
+    },
+  ],
+  gas: '200000',
+};
+const TEST_MEMO = `regen-js v${process.env.npm_package_version} tests`;
+
+const connect = async (): Promise<RegenApi> => {
+  const mnemonic =
+    'time dice choose cabbage suit panic silly cattle picture auto grab hole';
+
+  // This is an Amino signer
+  const signer = await Secp256k1HdWallet.fromMnemonic(mnemonic, {
+    prefix: 'regen',
+  });
+
+  return RegenApi.connect({
+    connection: {
+      type: 'tendermint',
+      endpoint: REDWOOD_NODE_TM_URL,
+      signer,
+    },
+  });
+};
+
+describe('RegenApi with tendermint connection', () => {
+  xdescribe('Signing and broadcasting Ecocredit txs', () => {
+    it('should sign and broadcast MsgSend using legacy amino sign mode', async () => {
+      let txRes: DeliverTxResponse | undefined;
+      const { msgClient } = await connect();
+      const TEST_MSG_SEND = MsgSend.fromPartial({
+        sender: TEST_ADDRESS,
+        recipient: TEST_ADDRESS,
+        credits: [
+          {
+            batchDenom: 'C01-001-20170606-20210601-007',
+            tradableAmount: '0.01',
+          },
+        ],
+      });
+
+      const signedTxBytes = await msgClient?.sign(
+        TEST_ADDRESS,
+        [TEST_MSG_SEND],
+        TEST_FEE,
+        TEST_MEMO,
+      );
+
+      expect(signedTxBytes).toBeTruthy();
+      if (signedTxBytes) {
+        txRes = await msgClient?.broadcast(signedTxBytes);
+        expect(txRes).toBeTruthy();
+        expect(txRes?.code).toBe(0);
+      }
+    });
+    it('should sign and broadcast MsgCreateClass using legacy amino sign mode', async () => {
+      let txRes: DeliverTxResponse | undefined;
+      const { msgClient } = await connect();
+      const TEST_MSG_CREATE_CLASS = MsgCreateClass.fromPartial({
+        admin: TEST_ADDRESS,
+        issuers: [TEST_ADDRESS],
+        metadata: 'unit test metadata',
+        creditTypeAbbrev: 'C',
+        fee: {
+          denom: 'uregen',
+          amount: '20000000',
+        },
+      });
+
+      const signedTxBytes = await msgClient?.sign(
+        TEST_ADDRESS,
+        [TEST_MSG_CREATE_CLASS],
+        TEST_FEE,
+        TEST_MEMO,
+      );
+
+      expect(signedTxBytes).toBeTruthy();
+      if (signedTxBytes) {
+        txRes = await msgClient?.broadcast(signedTxBytes);
+        expect(txRes).toBeTruthy();
+        expect(txRes?.code).toBe(0);
+      }
+    });
+  });
+  describe('Signing and broadcasting Basket txs', () => {
+    it('should sign and broadcast MsgCreate using legacy amino sign mode', async () => {
```

Comment:
I did some manual overriding of the amino conversions here and was able to get the tests to pass successfully.  Basically, the amino converters for this msg type need to be updated. Default values in the amino converters need to be set to `undefined` or the field must be removed from the JSON entirely. By default values, I'm mainly talking about `false`, `0`, `""`.  This is true also for nested messages.

Currently the basket msg converter isn't doing this correctly for `disable_auto_retire` or `date_criteria` fields.

Since `date_criteria` is itself an object with default values, there are two ways we could serialize it:
- either the amino converter creates an object with `min_state_date`, `min_date_window`, `years_in_the_past` fields -- all of them set to `undefined`
- OR the amino converter sets `date_criteria` to an empty JSON object `{}`

In either of these cases, the stringified value for `date_criteria` will be `{}` as `JSON.Stringify` removes `undefined` fields entirely.

See below for what the amino converter currently does, and what it needs to do to get the tests to pass:

```
    Generated SignDoc: {
      curator: 'regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp',
      name: 'TEST633',
      description: 'test description',
      exponent: 7,
      disable_auto_retire: false,
      credit_type_abbrev: 'C',
      allowed_classes: [ 'C14', 'C13' ],
      date_criteria: {
        min_start_date: undefined,
        start_date_window: undefined,
        years_in_the_past: 0
      },
      fee: [ { amount: '20000000', denom: 'uregen' } ]
    }
    -------------------------------------------------------
    -------------------------------------------------------
    Required SignDoc: {
      curator: 'regen1m0qh5y4ejkz3l5n6jlrntxcqx9r0x9xjv4vpcp',
      name: 'TEST633',
      description: 'test description',
      exponent: 7,
      disable_auto_retire: undefined,
      credit_type_abbrev: 'C',
      allowed_classes: [ 'C14', 'C13' ],
      date_criteria: {
        min_start_date: undefined,
        start_date_window: undefined,
        years_in_the_past: undefined
      },
      fee: [ { amount: '20000000', denom: 'uregen' } ]
    }
```

Reply:
Added a fix to the basket MsgCreate amino signers in #43